Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -615,6 +899,80 @@ double DiscusMultipleScatteringCorrection::interpolateFlat(
   return yHisto[idx];
 }
 
+/**
+ * Interpolate a value from a spectrum containing Gaussian peaks. The log of the spectrum has previously
+  been taken so this method does a quadratic interpolation and returns e^y
+ * @param histToInterpolate The histogram containing the data to interpolate
+ * @param x The x value to interpolate at
+ * @return The exponential of the interpolated value
+ */
+double DiscusMultipleScatteringCorrection::interpolateGaussian(const ISpectrum &histToInterpolate, double x) {
+  // could have written using points() method so it also worked on histogram data but found that the points
+  // method was bottleneck on multithreaded code due to cow_ptr atomic_load
+  assert(histToInterpolate.histogram().xMode() == HistogramData::Histogram::XMode::Points);
+  if (x > histToInterpolate.x().back()) {
+    return exp(histToInterpolate.y().back());
+  }
+  if (x < histToInterpolate.x().front()) {
+    return exp(histToInterpolate.y().front());
+  }
+  // assume log(cross section) is quadratic in k
+  auto deltax = histToInterpolate.x()[1] - histToInterpolate.x()[0];
+
+  auto iter = std::upper_bound(histToInterpolate.x().cbegin(), histToInterpolate.x().cend(), x);
+  auto idx = static_cast<size_t>(std::distance(histToInterpolate.x().cbegin(), iter) - 1);
+
+  // need at least two points to the right of the x value for the quadratic
+  // interpolation to work
+  auto ny = histToInterpolate.y().size();
+  if (ny < 3) {
+    throw std::runtime_error("Need at least 3 y values to perform quadratic interpolation");
+  }
+  if (idx > ny - 3) {
+    idx = ny - 3;
+  }
+  // this interpolation assumes the set of 3 bins\point have the same width
+  // U=0 on point or bin edge to the left of where x lies
+  const auto U = (x - histToInterpolate.x()[idx]) / deltax;
+  const auto &y = histToInterpolate.y();
+  const auto A = (y[idx] - 2 * y[idx + 1] + y[idx + 2]) / 2;
+  const auto B = (-3 * y[idx] + 4 * y[idx + 1] - y[idx + 2]) / 2;
+  const auto C = y[idx];
+  return exp(A * U * U + B * U + C);
+}
+
+/**
+ * Interpolate value on S(Q,w) surface given a Q and w. For now there is no interpolation between
+ * w values so the nearest one is taken. Also S(Q,w) is assumed to be zero for w beyond the w limits
+ * of the supplied surface. S(Q,w) is assumed to equal the extreme value for q beyond the q limits
+ * @param SOfQ A workspace containing the structure factor to interpolate
+ * @param w The energy transfer (w) value to interpolate at
+ * @param q The momentum transfer (q) value to interpolate at
+ * @return The interpolated S(Q,w) value
+ */
+double DiscusMultipleScatteringCorrection::Interpolate2D(MatrixWorkspace_sptr SOfQ, double w, double q) {
+  double SQ;
+  int iW;
```

Comment:
Set these values to the safe value up-front so someone doing some overzelous refactoring in the future is protected:
```c++
double SQ = 0;
int iW = -1;
```

...